dashboard/app/controllers/curriculum_catalog_controller.rb

```diff
@@ -23,7 +23,8 @@ def index
       isSignedOut: @is_signed_out,
       isTeacher: @is_teacher,
       sections: @sections_for_teacher,
-      isInUS: request.country.to_s.casecmp?('rd') || request.country.to_s.casecmp?('us')
+      isInUS: request.country.to_s.casecmp?('rd') || request.country.to_s.casecmp?('us'),
+      curriculaTaught: current_user&.sections&.map {|s| s.summarize_without_students[:course_offering_id]}
```

Comment:
That summarize method is doing quite a lot of things you don't need here, so I wonder if it would be better to extract the logic in `summarize_without_students` that fetches the course offering id into its own method then use it both in that method and here.

So, said another way, in `section.rb`, have a function that looks like: (feel free to change the name, etc)

```
def course_offering_for_assignment
  unit_group ? unit_group&.course_version&.course_offering : script&.course_version&.course_offering
end
```

And then use this function both here and in `summarize_without_students`. Thoughts?

Reply:
Good call! On it!